mod/groups/start.php

```diff
@@ -680,41 +695,6 @@ function groups_get_invited_groups($user_guid, $return_guids = false, $options =
 }
 
 /**
- * Join a user to a group, add river event, clean-up invitations
- *
- * @param ElggGroup $group
- * @param ElggUser  $user
- * @return bool
- */
-function groups_join_group($group, $user) {
-
-	// access ignore so user can be added to access collection of invisible group
-	$ia = elgg_set_ignore_access(TRUE);
```

Comment:
Reason is the hidden groups.... https://github.com/Elgg/Elgg/issues/1926 describes the issue... could you test if that is no longer an issue?